src/Chapter3.hs

```diff
@@ -970,6 +1067,19 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Week = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday
+  deriving (Show, Eq, Bounded, Enum)
+
+isWeekend :: Week -> Bool
+isWeekend day = day == Saturday || day == Sunday
+
+nextDay :: Week -> Week
+nextDay Sunday = minBound
+nextDay day = succ day
+
+daysToParty :: Week -> Int
+daysToParty Friday = 0
+daysToParty day    = (+1) . daysToParty . nextDay $ day
```

Comment:
Creative recursive solution 🧠 
You can also solve it using the `Enum` instance of your data type and `mod` function (we know that it is 7 days a week 🙂 )